src/osd/PG.h

```diff
@@ -826,7 +826,7 @@ class PG : public DoutPrefixProvider {
 	if (j == needs_recovery_map.end()) {
 	  needs_recovery_map.insert(*i);
 	} else {
-	  lgeneric_dout(pg->cct, 0) << this << " " << pg->info.pgid << " unexpected need for "
+	  lgeneric_dout(pg->cct, 15) << this << " " << pg->info.pgid << " redundant need for "
```

Comment:
That warning may be able to tell us what went wrong when we actually end up hitting the following assert? I agree, unexpected might not be the best word to describe other scenarios.